### ISSUE TITLE
Wait for functionality

### ISSUE THREAD

**Issue by ostap0207** [LukasBombach/Lazyloader#3]
Right now I have to load file and add callback at the same place.
Would be good to have an option just to wait for some file which has been loaded before with LazyLoader. For example:

One file:
`LazyLoader.load("/bar.css`)'

Another file:
`LazyLoader.waitFor("/bar.css", function() { //file is loaded})`

First idea the comes to mind is to add a callback by default to `load` that would mark the link with `data-loaded="true"`. And then poll data this in `LazyLoader.waitFor`.

**Comment by LukasBombach** [comment]
My idea was to extend the promise API with a `thenLoad`method like this:

``` javascript
LazyLoader.load('file1.css').then(doSomething).thenLoad('file2.css');
```

would that be a thing that fits your needs or did I get you wrong?

**Comment by ostap0207** [comment]
Not really. In your case you need to do the loading and provide callback in one file (one call even).
In my case I want to be able to load css in js one file. But check that it's loaded in a different file. I can help with pull request if needed.

**Comment by LukasBombach** [comment]
Ah, I get what you mean! Lemme think about how to do it!

**Comment by LukasBombach** [comment]
I plan to do something like this:

``` javascript
LazyLoader.onLoad("/bar.css", function(){})
LazyLoader.onLoad("/bar.css").then(function(){}) // both should work
```

This should work with files loaded by the lazy loader but also with any other js or css file included on the page. I may be able to do it on the weekend, but for this I will do some internal restructuring that I had planned anyway.

**Comment by ostap0207** [comment]
That's basically it. So if the file already included it doesn't add it again. Thanks.

**Comment by LukasBombach** [comment]
👍
